### PR TITLE
Implement fast register for pod tasks v2

### PR DESCRIPTION
**Pull request by katrogan**
Signed-off-by: Katrina Rogan <katroganGH@gmail.com>

# TL;DR
Pod tasks now write the pod spec to the task target directly. In order to fast-execute them, we must modify the primary container command to first run `pyflyte-fast-execute...`

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
User reported bug (thank you @jeevb)

## Tracking Issue
https://github.com/flyteorg/flyte/issues/1050

## Follow-up issue
_NA_

### CODE REVIEW

File: flytekit/core/python_auto_container.py [hunk 1]
```diff
@@ -141,24 +157,6 @@ def get_container(self, settings: SerializationSettings) -> _task_model.Containe
         )
 
 
-def _default_command(task: PythonAutoContainerTask, settings: SerializationSettings) -> List[str]:
```
Comment (wild-endeavor) [inline flytekit/core/python_auto_container.py:144]: why did you move this back into the class?

Comment (katrogan) [reply flytekit/core/python_auto_container.py:144]: so I can call <task>.get_default_command() for fast serialize. seemed more oop-y this way

File: flytekit/common/translator.py [hunk 2]
```diff
@@ -62,6 +62,24 @@ def to_serializable_cases(
...
+
+    def fn(settings: SerializationSettings) -> List[str]:
+        return [
+            "pyflyte-fast-execute",
+            "--additional-distribution",
+            "{{ .remote_package_path }}",
+            "--dest-dir",
+            "{{ .dest_dir }}",
+            "--",
+        ] + default_command
```
Comment (jeevb) [inline flytekit/common/translator.py:78]:
```suggestion
            "--",
            *default_command,
        ]
```

**Review by jeevb — APPROVED** [review]
LGTM!

File: flytekit/common/translator.py (hunk 2 above)
Comment (katrogan) [reply flytekit/common/translator.py]: done

**Review by wild-endeavor — APPROVED** [review]
